src/gpio.rs

```diff
@@ -1217,6 +1428,12 @@ mod chip {
 
     use super::*;
 
+    #[cfg(feature = "std")]
```

Comment:
What about `GpioPin`? Can it have subscription capabilities as well?

Reply:
I am not sure to understand how the generic GpioPin should be used.

Is this used as an alternative (though unsafe) method to get a gpio pin without taking it from Peripherals ? If so, I guess it should also have subscription capabilities.

Reply:
You can degrade/downgrade a normal pin into a GpioPin.
 This way you bo longer care in client code which pin exactly it was which means less generics typing, at the expense of carrying around the pin number.

Ideally, we should support subscribing to GpioPins too. Just wanted to make sure that's the case.